climpred/prediction.py

```diff
@@ -307,47 +196,40 @@ def compute_persistence(ds, reference, nlags=None, metric='pearson_r',
     Args:
         ds (xarray object): The initialized ensemble.
         reference (xarray object): The reference time series.
-        nlags (int, default None): Number of lags to compute persistence to.
-            If None, defaults to number of lead steps from the prediction
-            ensemble.
         metric (str): Metric name to apply at each lag for the persistence
                       computation. Default: 'pearson_r'
-        dim (str): Dimension over which to compute persistence forecast.
-                   Default: 'time'
 
     Returns:
         pers (xarray object): Results of persistence forecast with the input
                               metric applied.
     """
+    nlags = ds.lead.size
 
     check_xarray(reference)
-    if nlags is None:
-        nlags = ds.lead.size
 
     metric = get_metric_function(metric)
-    if metric not in [_pearson_r, _rmse, _mse, _mae]:
-        raise ValueError("""Please select between the following metrics:
-            'pearson_r',
-            'rmse',
-            'mse',
-            'mae'""")
+    if metric not in all_metrics:
+        raise ValueError("""Please select from all_metrics.'""")
```

Comment:
Same as above; maybe dump into a private function

`def _validate_metric():`

```
raise ValueError(f'Please input a metric from the following {ALL_METRIC_STRINGS}')
```

Reply:
different checks for PM and hindcast.